### PR TITLE
Add buildpack samples

### PR DESCRIPTION
**Pull request by JanakaSandaruwan**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: greeting-service-go/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+displayName: Greeting Service
+description: Responds with a record including the message received via an HTTP GET
+componentType: service
+buildPreset: go
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /greeting-service-go
+thumbnailPath: /greeting-service-go/icon.png
+documentationPath: /greeting-service-go/readme.md
```
Comment (a5anka) [inline greeting-service-go/metadata.yaml:8]:
```suggestion
documentationPath: /greeting-service-go/README.md
```

File: greeting-service-go/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+tags:
+- Category/Service
+- Language/Go
```
Comment (a5anka) [inline greeting-service-go/metadata.yaml:11]:
```suggestion
- Go
```

Comment (a5anka) [inline greeting-service-go/metadata.yaml:11]:
```suggestion
tags:
```

File: hello-world-go-task/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+description: Print Hello World message
+componentType: manual-task
+buildPreset: go
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /hello-world-go-task
+thumbnailPath: /hello-world-go-task/icon.png
+documentationPath: /hello-world-go-task/readme.md
+tags:
+- Category/Task
+- Language/Go
```
Comment (a5anka) [inline hello-world-go-task/metadata.yaml:11]:
```suggestion
tags:
```

File: hello-world-go-task/readme.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Hello World Go
+
+Sample for Hello World task. Print "Hello World" when run the main.go
+
+## Getting started
+
+1. Fork this repository.
+2. Delpoy as a manual task in Choreo.
```
Comment (a5anka) [inline hello-world-go-task/readme.md:8]: Shall we describe this more?

something similar to:
"Please refer to the Choreo documentation under the [Develop a REST API](https://wso2.com/choreo/docs/develop-components/develop-services/develop-a-rest-api/#step-1-create-a-service-component-from-a-dockerfile) section to learn how to deploy the application.

You can select either Docker or Go as buildpacks. Fill as follow according to selected Buildpack."

File: hello-world-go-webapp/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Hello World Go
+
+Sample for Hello World web app.
+
+## Getting started
+
+1. Fork this repository.
+2. Delpoy as a web app in Choreo.
```
Comment (a5anka) [inline hello-world-go-webapp/README.md:8]: Shall we describe this more like in greeting-service-go sample?

File: hello-world-go-webapp/metadata.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+description: A Hello World web app created using Go
+componentType: web-application
+buildPreset: go
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /hello-world-go-webapp
+thumbnailPath: /hello-world-go-webapp/icon.png
+documentationPath: /hello-world-go-webapp/readme.md
+tags:
+- Category/WebApplication
+- Language/Go
```
Comment (a5anka) [inline hello-world-go-webapp/metadata.yaml:11]:
```suggestion
tags:
- WebApplication
```

File: hello-world-java-task/metadata.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+description: Print Hello World message
+componentType: manual-task
+buildPreset: java
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /hello-world-java-task
+thumbnailPath: /hello-world-java-task/icon.png
+documentationPath: /hello-world-java-task/readme.md
+tags:
+- Category/Task
+- Language/Java
```
Comment (a5anka) [inline hello-world-java-task/metadata.yaml:11]:
```suggestion
tags:
```

File: hello-world-nodejs-task/metadata.yaml [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+description: Print Hello World message
+componentType: manual-task
+buildPreset: NodeJS
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /hello-world-nodejs-task
+thumbnailPath: /hello-world-nodejs-task/icon.png
+documentationPath: /hello-world-nodejs-task/readme.md
+tags:
+- Category/Task
+- Language/NodeJS
```
Comment (a5anka) [inline hello-world-nodejs-task/metadata.yaml:11]:
```suggestion
```

File: hello-world-nodejs-task/readme.md [hunk 9]
```diff
@@ -0,0 +1,8 @@
+# Hello World NodeJs
+
+Sample for Hello World task. Print "Hello World"
+
+## Getting started
+
+1. Fork this repository.
+2. Delpoy as a manual task in Choreo.
```
Comment (a5anka) [inline hello-world-nodejs-task/readme.md:8]: Shall we add more details?

File: hello-world-php-service/README.md [hunk 10]
```diff
@@ -0,0 +1,8 @@
+# Hello World PHP
+
+Sample for Hello World service.
+
+## Getting started
+
+1. Fork this repository.
+2. Delpoy as a service in Choreo.
```
Comment (a5anka) [inline hello-world-php-service/README.md:8]: Shall we add more details?

File: hello-world-php-service/metadata.yaml [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+description: A Hello World service created using Go
+componentType: service
+buildPreset: php
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /hello-world-php-service
+thumbnailPath: /hello-world-php-service/icon.png
+documentationPath: /hello-world-php-service/readme.md
+tags:
+- Category/Service
+- Language/PHP
```
Comment (a5anka) [inline hello-world-php-service/metadata.yaml:11]:
```suggestion
tags:
```

File: hello-world-php-task/metadata.yaml [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+description: Print Hello World message
+componentType: manual-task
+buildPreset: PHP
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /hello-world-php-task
+thumbnailPath: /hello-world-php-task/icon.png
+documentationPath: /hello-world-php-task/readme.md
+tags:
+- Category/Task
+- Language/PHP
```
Comment (a5anka) [inline hello-world-php-task/metadata.yaml:11]:
```suggestion
tags:
```

File: hello-world-php-task/readme.md [hunk 13]
```diff
@@ -0,0 +1,8 @@
+# Hello World PHP
+
+Sample for Hello World task. Print "Hello World"
+
+## Getting started
+
+1. Fork this repository.
+2. Delpoy as a manual task in Choreo.
```
Comment (a5anka) [inline hello-world-php-task/readme.md:8]: Shall we add more details?

File: hello-world-php-webapp/metadata.yaml [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+description: A Hello World web app created using PHP
+componentType: web-application
+buildPreset: php
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /hello-world-php-webapp
+thumbnailPath: /hello-world-php-webapp/icon.png
+documentationPath: /hello-world-php-webapp/readme.md
+tags:
+- Category/WebApplication
+- Language/PHP
```
Comment (a5anka) [inline hello-world-php-webapp/metadata.yaml:11]:
```suggestion
tags:
- WebApplication
```

File: hello-world-python-task/metadata.yaml [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+description: Print Hello World message
+componentType: manual-task
+buildPreset: python
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /hello-world-python-task
+thumbnailPath: /hello-world-python-task/icon.png
+documentationPath: /hello-world-python-task/readme.md
+tags:
+- Category/Task
+- Language/Python
```
Comment (a5anka) [inline hello-world-python-task/metadata.yaml:11]:
```suggestion
tags:
```

File: hello-world-python-task/readme.md [hunk 16]
```diff
@@ -0,0 +1,8 @@
+# Hello World Python
+
+Sample for Hello World task. Print "Hello World" when run the main.py
+
+## Getting started
+
+1. Fork this repository.
+2. Delpoy as a manual task in Choreo.
```
Comment (a5anka) [inline hello-world-python-task/readme.md:8]: Shall we add more details?

File: hello-world-ruby-service/README.md [hunk 17]
```diff
@@ -0,0 +1,8 @@
+# Hello World Ruby
+
+Sample for Hello World service. This API responds with a "Hello World" message when called, making it an excellent starting point for beginners and a handy tool for testing APIs with Ruby.
+
+## Getting started
+
+1. Fork this repository.
+2. Delpoy as a service in Choreo.
```
Comment (a5anka) [inline hello-world-ruby-service/README.md:8]: Shall we add more details?

File: hello-world-ruby-service/metadata.yaml [hunk 18]
```diff
@@ -0,0 +1,11 @@
...
+description: Return Hello World message
+componentType: service
+buildPreset: ruby
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /hello-world-ruby-service
+thumbnailPath: /hello-world-ruby-service/icon.png
+documentationPath: /hello-world-ruby-service/readme.md
+tags:
+- Category/Service
+- Language/Ruby
```
Comment (a5anka) [inline hello-world-ruby-service/metadata.yaml:11]:
```suggestion
tags:
```

File: hello-world-ruby-task/metadata.yaml [hunk 19]
```diff
@@ -0,0 +1,11 @@
...
+description: Print Hello World message
+componentType: manual-task
+buildPreset: ruby
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /hello-world-ruby-task
+thumbnailPath: /hello-world-ruby-task/icon.png
+documentationPath: /hello-world-ruby-task/readme.md
+tags:
+- Category/Task
+- Language/Ruby
```
Comment (a5anka) [inline hello-world-ruby-task/metadata.yaml:11]:
```suggestion
tags:
```

File: hello-world-ruby-task/readme.md [hunk 20]
```diff
@@ -0,0 +1,8 @@
+# Hello World Ruby
+
+Sample for Hello World task. Print "Hello World"
+
+## Getting started
+
+1. Fork this repository.
+2. Delpoy as a manual task in Choreo.
```
Comment (a5anka) [inline hello-world-ruby-task/readme.md:8]: Shall we add more details?

File: hello-world-ruby-webapp/README.md [hunk 21]
```diff
@@ -0,0 +1,9 @@
+# Hello World Ruby
+
+Sample for Hello World web app.
+
+## Getting started
+
+1. Fork this repository.
+2. Delpoy as a web app in Choreo.
```
Comment (a5anka) [inline hello-world-ruby-webapp/README.md:8]: Shall we add more details?

File: hello-world-ruby-webapp/metadata.yaml [hunk 22]
```diff
@@ -0,0 +1,11 @@
...
+description: A Hello World web app created using Ruby
+componentType: web-application
+buildPreset: ruby
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /hello-world-ruby-webapp
+thumbnailPath: /hello-world-ruby-webapp/icon.png
+documentationPath: /hello-world-ruby-webapp/readme.md
+tags:
+- Category/WebApplication
+- Language/Ruby
```
Comment (a5anka) [inline hello-world-ruby-webapp/metadata.yaml:11]:
```suggestion
tags:
- WebApplication
```

File: product-management-service/metadata.yaml [hunk 23]
```diff
@@ -0,0 +1,11 @@
...
+description: CRUD operations for products
+componentType: service
+buildPreset: java
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /product-management-service
+# thumbnailPath: /product-management-service/icon.png
+documentationPath: /product-management-service/readme.md
+tags:
+- Category/Service
+- Language/Java
```
Comment (a5anka) [inline product-management-service/metadata.yaml:11]:
```suggestion
tags:
```

File: reading-books-list-service-nodejs/metadata.yaml [hunk 24]
```diff
@@ -0,0 +1,11 @@
...
+description: CRUD operations for books
+componentType: service
+buildPreset: nodejs
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /reading-books-list-service-nodejs
+# thumbnailPath: /reading-books-list-service-nodejs/icon.png
+documentationPath: /reading-books-list-service-nodejs/readme.md
+tags:
+- Category/Service
+- Language/NodeJS
```
Comment (a5anka) [inline reading-books-list-service-nodejs/metadata.yaml:11]:
```suggestion
tags:
```

File: reading-books-list-service-python/metadata.yaml [hunk 25]
```diff
@@ -0,0 +1,11 @@
...
+description: CRUD operations for books
+componentType: service
+buildPreset: python
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /reading-books-list-service-python
+# thumbnailPath: /reading-books-list-service/icon.png
+documentationPath: /reading-books-list-service-python/readme.md
+tags:
+- Category/Service
+- Language/Python
```
Comment (a5anka) [inline reading-books-list-service-python/metadata.yaml:11]:
```suggestion
tags:
```

File: greeting-service-go/metadata.yaml (hunk 1 above)
Comment (JanakaSandaruwan) [reply greeting-service-go/metadata.yaml:8]: The correct file name is `readme.md`

File: reading-books-list-service-nodejs/metadata.yaml [hunk 26]
```diff
@@ -0,0 +1,11 @@
+displayName: Book Management Service in Java
+description: CRUD operations for books
+componentType: service
+buildPreset: nodejs
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /reading-books-list-service-nodejs
+# thumbnailPath: /reading-books-list-service-nodejs/icon.png
```
Comment (chameerar) [inline reading-books-list-service-nodejs/metadata.yaml:7]: Shall we add an icon for missing samples? We can use this icon for the time being - https://github.com/wso2/choreo-samples/blob/main/jwt-handling/icon.png

Please check on other places too.

File: product-management-service/metadata.yaml [hunk 27]
```diff
@@ -0,0 +1,11 @@
+displayName: Product Management Service in Java
+description: CRUD operations for products
+componentType: service
+buildPreset: java
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /product-management-service
+# thumbnailPath: /product-management-service/icon.png
+documentationPath: /product-management-service/readme.md
```
Comment (chameerar) [inline product-management-service/metadata.yaml:8]:
```suggestion
documentationPath: /product-management-service/README.md
```

File: hello-world-ruby-webapp/metadata.yaml [hunk 28]
```diff
@@ -0,0 +1,11 @@
+displayName: Hello World Web in Ruby
+description: A Hello World web app created using Ruby
+componentType: web-application
+buildPreset: ruby
+repositoryUrl: https://github.com/wso2/choreo-samples/
+componentPath: /hello-world-ruby-webapp
+thumbnailPath: /hello-world-ruby-webapp/icon.png
+documentationPath: /hello-world-ruby-webapp/readme.md
```
Comment (chameerar) [inline hello-world-ruby-webapp/metadata.yaml:8]: `readme.md` should be changed to `README.md`. 
Please check readme file path in other places too.

File: greeting-service-go/metadata.yaml (hunk 2 above)
Comment (JanakaSandaruwan) [reply greeting-service-go/metadata.yaml]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/add09046e6b8c2d071d9af8ab6d9b1722a30038e

File: hello-world-go-task/metadata.yaml (hunk 3 above)
Comment (JanakaSandaruwan) [reply hello-world-go-task/metadata.yaml]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/add09046e6b8c2d071d9af8ab6d9b1722a30038e

File: hello-world-go-task/readme.md (hunk 4 above)
Comment (JanakaSandaruwan) [reply hello-world-go-task/readme.md]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/add09046e6b8c2d071d9af8ab6d9b1722a30038e

File: hello-world-go-webapp/README.md (hunk 5 above)
Comment (JanakaSandaruwan) [reply hello-world-go-webapp/README.md]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/add09046e6b8c2d071d9af8ab6d9b1722a30038e

File: hello-world-nodejs-task/readme.md (hunk 9 above)
Comment (JanakaSandaruwan) [reply hello-world-nodejs-task/readme.md]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/add09046e6b8c2d071d9af8ab6d9b1722a30038e

File: hello-world-php-service/README.md (hunk 10 above)
Comment (JanakaSandaruwan) [reply hello-world-php-service/README.md]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/add09046e6b8c2d071d9af8ab6d9b1722a30038e

File: hello-world-php-task/readme.md (hunk 13 above)
Comment (JanakaSandaruwan) [reply hello-world-php-task/readme.md]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/add09046e6b8c2d071d9af8ab6d9b1722a30038e

File: hello-world-python-task/readme.md (hunk 16 above)
Comment (JanakaSandaruwan) [reply hello-world-python-task/readme.md]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/add09046e6b8c2d071d9af8ab6d9b1722a30038e

File: hello-world-ruby-service/README.md (hunk 17 above)
Comment (JanakaSandaruwan) [reply hello-world-ruby-service/README.md]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/add09046e6b8c2d071d9af8ab6d9b1722a30038e

File: hello-world-ruby-task/readme.md (hunk 20 above)
Comment (JanakaSandaruwan) [reply hello-world-ruby-task/readme.md]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/add09046e6b8c2d071d9af8ab6d9b1722a30038e

File: hello-world-ruby-webapp/README.md (hunk 21 above)
Comment (JanakaSandaruwan) [reply hello-world-ruby-webapp/README.md]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/add09046e6b8c2d071d9af8ab6d9b1722a30038e

File: hello-world-ruby-webapp/metadata.yaml (hunk 28 above)
Comment (JanakaSandaruwan) [reply hello-world-ruby-webapp/metadata.yaml:8]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/a67aed655be39d5f54333c8866d56f03c6d43989

File: reading-books-list-service-nodejs/metadata.yaml (hunk 26 above)
Comment (JanakaSandaruwan) [reply reading-books-list-service-nodejs/metadata.yaml]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/a67aed655be39d5f54333c8866d56f03c6d43989

File: reading-books-list-service-python/metadata.yaml (hunk 25 above)
Comment (JanakaSandaruwan) [reply reading-books-list-service-python/metadata.yaml]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/a67aed655be39d5f54333c8866d56f03c6d43989

File: reading-books-list-service-nodejs/metadata.yaml (hunk 24 above)
Comment (JanakaSandaruwan) [reply reading-books-list-service-nodejs/metadata.yaml]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/a67aed655be39d5f54333c8866d56f03c6d43989

File: product-management-service/metadata.yaml (hunk 23 above)
Comment (JanakaSandaruwan) [reply product-management-service/metadata.yaml]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/a67aed655be39d5f54333c8866d56f03c6d43989

File: hello-world-ruby-task/metadata.yaml (hunk 19 above)
Comment (JanakaSandaruwan) [reply hello-world-ruby-task/metadata.yaml:11]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/a67aed655be39d5f54333c8866d56f03c6d43989

File: hello-world-go-webapp/metadata.yaml (hunk 6 above)
Comment (JanakaSandaruwan) [reply hello-world-go-webapp/metadata.yaml]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/a67aed655be39d5f54333c8866d56f03c6d43989

File: hello-world-nodejs-task/metadata.yaml (hunk 8 above)
Comment (JanakaSandaruwan) [reply hello-world-nodejs-task/metadata.yaml]: Fixed

File: hello-world-ruby-service/metadata.yaml (hunk 18 above)
Comment (JanakaSandaruwan) [reply hello-world-ruby-service/metadata.yaml]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/a67aed655be39d5f54333c8866d56f03c6d43989

File: hello-world-ruby-webapp/metadata.yaml (hunk 22 above)
Comment (JanakaSandaruwan) [reply hello-world-ruby-webapp/metadata.yaml]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/a67aed655be39d5f54333c8866d56f03c6d43989

File: product-management-service/metadata.yaml (hunk 27 above)
Comment (JanakaSandaruwan) [reply product-management-service/metadata.yaml:8]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/a67aed655be39d5f54333c8866d56f03c6d43989

File: hello-world-ruby-task/metadata.yaml (hunk 19 above)
Comment (JanakaSandaruwan) [reply hello-world-ruby-task/metadata.yaml:11]: Fixed with https://github.com/wso2/choreo-samples/pull/15/commits/711b25893cce0137110549598129d0758fd9441c

File: hello-world-php-task/metadata.yaml (hunk 12 above)
Comment (JanakaSandaruwan) [reply hello-world-php-task/metadata.yaml:11]: Fixed

File: hello-world-java-task/metadata.yaml (hunk 7 above)
Comment (JanakaSandaruwan) [reply hello-world-java-task/metadata.yaml]: Fixed

**Review by a5anka — APPROVED** [review]

**Pull request merged**